includes/Core/Storage/Setting_With_Owned_Keys_Trait.php

```diff
@@ -51,21 +49,53 @@ function ( $option, $value ) use ( $keys ) {
 
 		add_filter(
 			'pre_update_option_' . static::OPTION,
-			function( $value, $old_value ) use ( $keys ) {
+			function ( $value, $old_value ) {
 				if ( is_array( $value ) && is_array( $old_value ) ) {
-					foreach ( $keys as $key ) {
-						if ( isset( $value[ $key ], $old_value[ $key ] ) && $value[ $key ] !== $old_value[ $key ] && current_user_can( Permissions::MANAGE_OPTIONS ) ) {
-							$value['ownerID'] = get_current_user_id();
-							break;
-						}
-					}
+					return $this->maybe_update_owner_id_in_settings_before_saving_them( $value, $old_value );
 				}
-
-				return $value;
 			},
 			10,
 			2
 		);
 	}
 
+	/**
+	 * Updates settings to set the current user as an owner of the module if settings contain owned keys.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The module settings.
+	 */
+	protected function maybe_set_owner_id( $settings ) {
```

Comment:
As methods provided by the `Setting_With_Owned_Keys_Trait`, they should probably be named a bit more specifically to "owned keys".

Maybe something like `maybe_set_initial_owner_id`?

Reply:
Renamed it.